### ISSUE TITLE
NX CLOUD ERROR - This version of Nx Cloud is incompatible with the @nrwl/* and @nx/* packages in your workspace, or Nx was not installed properly.

### ISSUE THREAD

**Issue by C0ZEN** [nrwl/nx#17175]
### Current Behavior

In my GitHub CI, every run on a pull request is working as expected.
It was configured by following the [Distributed CI with Nx Cloud](https://nx.dev/recipes/ci/monorepo-ci-github-actions#distributed-ci-with-nx-cloud).

Yet, for some weeks now, all the runs are failing on the develop branch (see the Failure Logs section for more details).

I'm currently using the version 16.2.1 of Nx packages.

### Expected Behavior

Since my project is using Nx 16, and the CI is using the latest (16 as well) version of Nx Cloud, this error message makes no sense to me, and I expect my build to pass.
I don't get why on a pull request everything is fine, but on the develop branch it's failing.

### Steps to Reproduce

None

### Nx Report

```shell
None
```

### Failure Logs

```shell
npm WARN exec The following package was not found and will be installed: nx-cloud@16.0.5
NX CLOUD ERROR
---------------------------------------
This version of Nx Cloud is incompatible with the @nrwl/* and @nx/* packages in your workspace, or Nx was not installed properly.

Verify your install step was successful, and if it was,
match your @nrwl/nx-cloud version to the same major version of your @nx/* and @nrwl/* packages and try again.
---------------------------------------
Error: Process completed with exit code 1.
```

### Operating System

- [ ] macOS
- [X] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by StalkAltan** [comment]
Hi @C0ZEN 

Can you provide logs to confirm that the install step was successful in your pipeline? Often, when this errors, the logs this is further up in the GitHub UI than this Nx Cloud error, which doesn't make it particularly easy to spot at first glance.

If it turns out that is successful we can diagnose further.

Thanks!

**Comment by C0ZEN** [comment]
@StalkAltan thank you for taking the time here.
There is indeed another error further up in the pipeline.

```
2023-05-24T08:21:19.5435594Z [command]/usr/bin/git branch --list --remote origin/dev
2023-05-24T08:21:19.5522773Z   origin/dev
2023-05-24T08:21:19.5529205Z [command]/usr/bin/git rev-parse refs/remotes/origin/dev
2023-05-24T08:21:19.5599546Z 454d9a83c2f4ba95e8678c9d73c74994fe328b24
2023-05-24T08:21:19.5624592Z ##[endgroup]
2023-05-24T08:21:19.5625792Z ##[group]Determining the checkout info
2023-05-24T08:21:19.5626568Z ##[endgroup]
2023-05-24T08:21:19.5627195Z ##[group]Checking out the ref
2023-05-24T08:21:19.5628014Z [command]/usr/bin/git checkout --progress --force -B dev refs/remotes/origin/dev
2023-05-24T08:21:20.0529006Z Switched to a new branch 'dev'
2023-05-24T08:21:20.0529807Z branch 'dev' set up to track 'origin/dev'.
2023-05-24T08:21:20.0556663Z ##[endgroup]
2023-05-24T08:21:20.0662801Z [command]/usr/bin/git log -1 --format='%H'
2023-05-24T08:21:20.0756314Z '454d9a83c2f4ba95e8678c9d73c74994fe328b24'
2023-05-24T08:21:20.1127156Z ##[group]Run nrwl/nx-set-shas@v3
2023-05-24T08:21:20.1127461Z with:
2023-05-24T08:21:20.1127677Z   main-branch-name: dev
2023-05-24T08:21:20.1127989Z   set-environment-variables-for-job: true
2023-05-24T08:21:20.1128407Z   error-on-no-successful-workflow: false
2023-05-24T08:21:20.1128841Z   last-successful-event: push
2023-05-24T08:21:20.1129241Z   working-directory: .
2023-05-24T08:21:20.1129637Z env:
2023-05-24T08:21:20.1129987Z   NX_CLOUD_DISTRIBUTED_EXECUTION: true
2023-05-24T08:21:20.1130480Z   NX_CLOUD_DISTRIBUTED_EXECUTION_AGENT_COUNT: 3
2023-05-24T08:21:20.1130889Z   NX_BRANCH: dev
2023-05-24T08:21:20.1131232Z   NX_CLOUD_ACCESS_TOKEN: 
2023-05-24T08:21:20.1131587Z   NX_CLOUD_AUTH_TOKEN: 
2023-05-24T08:21:20.1131933Z   NPM_TOKEN: 
2023-05-24T08:21:20.1132631Z   WORKLOAD_IDENTITY_PROVIDER: ***
2023-05-24T08:21:20.1133362Z   SERVICE_ACCOUNT: ***
2023-05-24T08:21:20.1133712Z ##[endgroup]
2023-05-24T08:21:20.1512679Z ##[group]Run node $GITHUB_ACTION_PATH/dist/index.js *** dev false push . 
2023-05-24T08:21:20.1513584Z [36;1mnode $GITHUB_ACTION_PATH/dist/index.js *** dev false push . [0m
2023-05-24T08:21:20.2444577Z shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
2023-05-24T08:21:20.2445387Z env:
2023-05-24T08:21:20.2445793Z   NX_CLOUD_DISTRIBUTED_EXECUTION: true
2023-05-24T08:21:20.2446471Z   NX_CLOUD_DISTRIBUTED_EXECUTION_AGENT_COUNT: 3
2023-05-24T08:21:20.2446988Z   NX_BRANCH: dev
2023-05-24T08:21:20.2447431Z   NX_CLOUD_ACCESS_TOKEN: 
2023-05-24T08:21:20.2447834Z   NX_CLOUD_AUTH_TOKEN: 
2023-05-24T08:21:20.2448278Z   NPM_TOKEN: 
2023-05-24T08:21:20.2448975Z   WORKLOAD_IDENTITY_PROVIDER: ***
2023-05-24T08:21:20.2449656Z   SERVICE_ACCOUNT: ***
2023-05-24T08:21:20.2450060Z ##[endgroup]
2023-05-24T08:21:20.9905563Z ##[error]Resource not accessible by integration
2023-05-24T08:21:21.0993923Z ##[error]Process completed with exit code 1.
2023-05-24T08:21:21.1220821Z ##[group]Run npx nx-cloud@16 stop-all-agents
2023-05-24T08:21:21.1221396Z [36;1mnpx nx-cloud@16 stop-all-agents[0m
2023-05-24T08:21:21.1991295Z shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
2023-05-24T08:21:21.1991835Z env:
2023-05-24T08:21:21.1992269Z   NX_CLOUD_DISTRIBUTED_EXECUTION: true
2023-05-24T08:21:21.1992760Z   NX_CLOUD_DISTRIBUTED_EXECUTION_AGENT_COUNT: 3
2023-05-24T08:21:21.1993264Z   NX_BRANCH: dev
2023-05-24T08:21:21.1993653Z   NX_CLOUD_ACCESS_TOKEN: 
2023-05-24T08:21:21.1993992Z   NX_CLOUD_AUTH_TOKEN: 
2023-05-24T08:21:21.1994354Z   NPM_TOKEN: 
2023-05-24T08:21:21.1995113Z   WORKLOAD_IDENTITY_PROVIDER: ***
2023-05-24T08:21:21.1995818Z   SERVICE_ACCOUNT: ***
2023-05-24T08:21:21.1996197Z ##[endgroup]
2023-05-24T08:21:22.9300765Z npm WARN exec The following package was not found and will be installed: nx-cloud@16.0.5
2023-05-24T08:21:25.2297844Z NX CLOUD ERROR
2023-05-24T08:21:25.2303706Z ---------------------------------------
2023-05-24T08:21:25.2304715Z This version of Nx Cloud is incompatible with the @nrwl/* and @nx/* packages in your workspace, or Nx was not installed properly.
2023-05-24T08:21:25.2305149Z 
2023-05-24T08:21:25.2305356Z Verify your install step was successful, and if it was,
2023-05-24T08:21:25.2306202Z match your @nrwl/nx-cloud version to the same major version of your @nx/* and @nrwl/* packages and try again.
2023-05-24T08:21:25.2306770Z ---------------------------------------
2023-05-24T08:21:25.2560140Z ##[error]Process completed with exit code 1.
```

Is it helping?

**Comment by StalkAltan** [comment]
@C0ZEN yes, I believe something in your additions has broken this. You can see that there is no install happening from your package manager.

You are receiving this error because the `npx nx-cloud stop-all-agents` command is set to *always* run. This is what will disable distributed agents in case another step in your pipeline fails. The Nx Cloud error here is a side effect of a previous step failing, rather than an issue with Nx Cloud.

**Comment by C0ZEN** [comment]
TL;DR: I fixed my issue 🚀 

I added the permission `actions: 'read'` by following the [nrwl/nx-set-shas documentation](https://github.com/nrwl/nx-set-shas#permissions-in-v2).
I also provided the _input_ `working-directory` to this action because in my case, my repository contains other projects, and so Nx is configured in a sub-folder.

```yml
permissions:
  id-token: write
  contents: write
  actions: read
...
- name: Derive appropriate SHAs for base and head for `nx affected` commands
  uses: nrwl/nx-set-shas@v3
  with:
    main-branch-name: ${{ inputs.main-branch-name }}
    working-directory: ${{ inputs.working-directory || github.workspace }}
```
